### ISSUE TITLE
Immediately scrolling down upon page load does not trigger fetch more

### ISSUE THREAD

**Issue by coolcomfort** [ankeetmaini/react-infinite-scroll-component#283]
I am using chrome, and when experimenting with infinite-scroll, I noticed that if one quickly scrolls to the bottom of a page, while refreshing, the page will start at the lowest scrollable point (using Chrome). When attempting this with infinite-scroll, I notice that infinite-scroll fails to fetch more, despite being at the lowest scrollable point when it should. This more than likely has to do with the fact that infinite scroll isn't aware the user is already at the bottom of the page. Is there any type of fix for this? I consider it UI breaking, because there are reasons for a user to quickly scroll through paginated results (i.e., a user was interrupted during his search and is trying to return to where he left off).

**Comment by Sooraj-s-98** [comment]
me also  facing same issue @coolcomfort

**Comment by coolcomfort** [comment]
I actually managed to find the solution, and it is not an issue with react-infinite-scroll. I will try to explain what happened. If you notice in the examples provided, this issue does not happen (https://codesandbox.io/s/yk7637p62z). To replicate it best, I recommend copying this over to a blank project and looking for yourself, as it is impossible to start at the lowest scrollable point in codesandbox due to multiple scrollable windows.

When integrating my code, I had tried to do it within the exact same state that was provided in the example. I am using apollo + graphql, so there were 2 clones of my data, the original apollo query grab, and the state that was cloning that. You do not want to duplicate your state this way, and you especially do not want to map over the duplicated state. Instead, you want to map over the original apollo query grab. 

Here is my example:
```
  const queryResult = useQuery(ALL_COMMENTS, {
    variables: {
      first,
      skip: 0,
    },
  });

  const dataComments = queryResult.data;
```
... some code later ...

```
  return (
      <InfiniteScroll
        dataLength={count - first}
        next={fetchMoreData}
        hasMore={true}
        loader={<h4>Loading...</h4>}
      >
        {dataComments.allComments.map((value, i) => (
          <div style={style} key={value.id}>
            {value.comment} - #{i}
          </div>
        ))}
      </InfiniteScroll>
    )
```

As mentioned, previously I tried to map over a cloned dataComments within state. Although this stays more aligned with the example provided, it fails. I'm not entirely sure why it fails, because if you console log the data you will see the cloned data has successfully fetched new the data, but for some reason, it fails to render on the page. Even if this was working, I think it is better to stick with the original data in order to prevent needless cloning. I hope this helped whoever needed it!
